helpdesk_mgmt/models/helpdesk_ticket.py

```diff
@@ -10,13 +10,45 @@ class HelpdeskTicket(models.Model):
     _mail_post_access = "read"
     _inherit = ["mail.thread.cc", "mail.activity.mixin", "portal.mixin"]
 
+    def _get_applicable_stages(self, team):
+        if team:
+            return team._determine_stage()[team.id].id
+        else:
+            return (
+                self.env["helpdesk.ticket.stage"]
+                .search([("team_ids", "=", False)], limit=1)
+                .id
+            )
+
     def _get_default_stage_id(self):
-        return self.env["helpdesk.ticket.stage"].search([], limit=1).id
+        return self._get_applicable_stages(self.team_id)
+
+    @api.depends("team_id")
+    def _compute_stage_ids(self):
+        for ticket in self:
+            ticket.stage_id = self._get_applicable_stages(ticket.team_id)
+
+    def _default_team_id(self):
+        team_id = (
```

Comment:
This should call to the default team instead.

Reply:
what are you trying to say?

Reply:
Instead of repeating the code for the default team, call the method for getting the default team